Code/GraphMol/MolDraw2D/MolDraw2D.h

```diff
@@ -79,12 +91,26 @@ class MolDraw2D {
       const std::map<int, double> *highlight_radii = NULL, int confId = -1);
 
   virtual void drawMolecule(
+      const ROMol &mol, const std::string &legend,
+      const std::vector<int> *highlight_atoms = NULL,
+      const std::map<int, DrawColour> *highlight_map = NULL,
+      const std::map<int, double> *highlight_radii = NULL, int confId = -1);
+
+  virtual void drawMolecule(
```

Comment:
It's a little bit tricky because we have the C++ library and 3 wrappers to consider.

In C++, the canonical way is to mark it as deprecated at compile time. C++14 has the [[deprecated]] attribute, but that's a long ways off.

http://stackoverflow.com/questions/295120/c-mark-as-deprecated

Although the SO example only lists GCC, it looks like it's also supported in Clang.

In Python, especially for those who are fetching it from conda or pre-built binaries where there isn't an explicit compile step, the only sane way is to throw a DeprecationWarning (https://docs.python.org/2/library/warnings.html) in the wrapper and hope that they're not suppressing stderr.

In Java and C#, I have no clue.

This is all complicated by the fact that you're overloading the functions... so you'd need to mark one particular signature as being deprecated (though this shouldn't be a big barrier).